Explorer/Assets/DCL/Character/CharacterMotion/Systems/TeleportPositionCalculationSystem.cs

```diff
@@ -0,0 +1,165 @@
+﻿using Arch.Core;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using DCL.CharacterCamera;
+using DCL.CharacterMotion.Components;
+using DCL.Diagnostics;
+using DCL.Ipfs;
+using DCL.RealmNavigation;
+using ECS.Abstract;
+using UnityEngine;
+using Utility;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine.Pool;
+using Random = System.Random;
+
+namespace DCL.Character.CharacterMotion.Systems
+{
+    [UpdateInGroup(typeof(PostRenderingSystemGroup))]
+    [LogCategory(ReportCategory.MOTION)]
+    public partial class TeleportPositionCalculationSystem : BaseUnityLoopSystem
+    {
+        private static readonly Random RANDOM = new ();
+
+        private readonly Entity playerEntity;
+
+        private SingleInstanceEntity? cameraCached;
+        private SingleInstanceEntity cameraEntity => cameraCached ??= World.CacheCamera();
+
+        public TeleportPositionCalculationSystem(World world, Entity playerEntity) : base(world)
+        {
+            this.playerEntity = playerEntity;
+        }
+
+        protected override void Update(float t)
+        {
+            ref PlayerTeleportIntent teleportIntent = ref World.TryGetRef<PlayerTeleportIntent>(playerEntity, out bool hasTeleportIntent);
```

Comment:
I would do Querys and avoid `TryGetRef` in the Update. Unless there is a performance consideration?

Reply:
as it was discussed, Query are less performant for one entity than directly getting component directly (due to all GetEnumarator logic for Chunks in Arch). I can move it to query since it is not performance critical code

Reply:
changed to Query